### DISCUSSION TITLE
Increasing Storage Size not working for Samsung Devices

### DISCUSSION THREAD

**Discussion by kutaisan** [react-native-async-storage/async-storage#608] [Q&A]
## Current behavior

 I set up new storage size to 20MB using this [guide](https://react-native-async-storage.github.io/async-storage/docs/advanced/db_size). It works for Xiaomi, Oppo etc devices however not working on Samsung devices. 

`Row too big to fit into CursorWindow requiredPos=0, totalRows=1`

I was getting this error before adding more spaces and it disappared after adding more spaces on my debug devices except Samsung. 
Is there any hardware-based solutions for this?

## Expected behavior

Works on Samsung devices too.

## Repro steps

- Use a data which contains data > 6MB
- Add more spaces using the guide
- Test Samsung and another device

## Environment

- **Platforms tested**:
  - [ x ] Android
  - [ ] iOS
  - [ ] macOS
  - [ ] Windows
- **AsyncStorage version**: 1.13.2
- **Environment**:
    System:
    OS: macOS 11.2.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8700B CPU @ 3.20GHz
    Memory: 77.17 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.0.1 - /usr/local/bin/node
    Yarn: Not Found
    npm: 7.0.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
    Android SDK:
      API Levels: 29, 30
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2
      System Images: android-29 | Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6858069
    Xcode: 12.2/12B45b - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_272 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.1 => 0.64.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

**Comment by kutaisan** [comment]
Why it's closing without giving any information?

**Reply by sasweb** [reply]
This looks very similar: https://github.com/react-native-async-storage/async-storage/issues/537
